resources/ssm-patchbaselines.go

```diff
@@ -0,0 +1,69 @@
+package resources
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+type SSMPatchBaseline struct {
+	svc *ssm.SSM
+	ID  *string
+}
+
+func init() {
+	register("SSMPatchBaseline", ListSSMPatchBaselines)
+}
+
+func ListSSMPatchBaselines(sess *session.Session) ([]Resource, error) {
+	svc := ssm.New(sess)
+	resources := []Resource{}
+
+	params := &ssm.DescribePatchBaselinesInput{
+		MaxResults: aws.Int64(50),
+	}
+
+	for {
+		output, err := svc.DescribePatchBaselines(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, baselineIdentity := range output.BaselineIdentities {
+			if baselineIdentity.BaselineDescription == nil {
+				resources = append(resources, &SSMPatchBaseline{
+					svc: svc,
+					ID:  baselineIdentity.BaselineId,
+				})
+			} else if !strings.Contains(*baselineIdentity.BaselineDescription, "Provided by AWS") {
```

Comment:
To filter resources I would generally recommend the `Filter()` function. E.g.: https://github.com/rebuy-de/aws-nuke/blob/f9a3a639fe63fc9be7060556f53335635ce5da74/resources/iam-role-policy-attachments.go#L53-L58

In this case, I actually think your approach in `SSMDocument` is really great. It reduces the number of resources returned and let's AWS do the filter work. I tried it briefly with `aws ssm describe-patch-baselines --filters="Key=OWNER,Values=Self"` and it looks promising. Wdyt?

Reply:
I have an issue with Patch Baselines, which I am not sure will be easy to resolve, you can't delete the default.  So either we accept that, or we make a ton of conditional coding to find the right AWS provided one which matches the right OS.  Thoughts?

Reply:
Wouldn't filtering by `Owner=Self` resolve that issue?